src/analyze/analyze-security.c

```diff
@@ -1782,15 +1789,15 @@ static int assess(const SecurityInfo *info,
                                 description = access_description_na(a, policy);
                                 color = NULL;
                         } else if (badness == a->range) {
-                                checkmark = special_glyph(SPECIAL_GLYPH_CROSS_MARK);
+                                checkmark = ((json_format_flags == JSON_FORMAT_OFF) ? special_glyph(SPECIAL_GLYPH_CROSS_MARK) : "no");
```

Comment:
as mentioned in my earlier review: please rework format-table.[ch] to get a new field table TABLE_BOOLEAN_CHECKMARK which would be very much like TABLE_BOOLEAN, but instead of displaying yes/no for booleans in console output, would chow cross mark/check mark instead — and real booleans in JSON mode. Then, rework this code here to use that instead of TABLE_STRING for the second column.

Or in othewords: please push the booleans here down into the table formatter, and add the logic to display the right glyphs for them to the table formatter code, instead of here, ahead of time.

Reply:
I'm not sure I completely understand. How would this new field table know when json_format_flags has been set to be JSON_FORMAT_OFF and when it is set with the other two values (pretty | short). Or are you looking for something like this:

    r = table_add_many(details_table,
                                           (json_format_flags == JSON_FORMAT_OFF) ? TABLE_BOOLEAN_CHECKMARK: 
                                            TABLE_BOOLEAN, checkmark,
                                           TABLE_SET_MINIMUM_WIDTH, 1,
                                            .....);

And what is checkmark's value set to in the if else statements above. The original before my change was: 
    
    else if (badness == a->range) {
                                checkmark = special_glyph(SPECIAL_GLYPH_CROSS_MARK);
                                ......
    }

Reply:
if you look in format-table.c you'll find two very distinct code paths: table_data_to_json() generates json objects from the table content. here the new TABLE_BOOLEAN_CHECKMARK should have the exact same codepaths as TABLE_BOOLEAN. And then there is table_data_format() generates text to output if text output is requested, here you should use different codepaths for TABLE_BOOLEAN and TABLE_BOOLEAN_CHECKMARK.

table_print_with_pager() already contains the right logic that dispatches things towards one of the codepaths mentioned above in JSON mode, and to the other in text mode, you don't have to take care of that anymore, it already gets a json_format_flags parameter to figure that out.